apps/camera/js/lib/camera/camera.js

```diff
@@ -981,77 +1000,60 @@ Camera.prototype.stoppedRecording = function(recorded) {
   this.orientation.start();
 
   var self = this;
-  var elapsedTime = self.get('videoElapsed');
-  var filepath;
-  var req;
-  var storage;
+  var videoReq;
+  var posterReq;
+  var video;
+  var got = 0;
+
   if (recorded) {
-    // Re-fetch the blob from storage
-    storage = this.storage.video;
-    filepath = this.video.filepath;
-    req = storage.get(filepath);
-    req.onerror = onError;
-    req.onsuccess = onSuccess;
+    video = {
+      filepath: this.video.filepath,
+      rotation: this.video.rotation,
+      width: this.video.width,
+      height: this.video.height,
+      poster: {
+        filepath: this.video.poster.filepath,
+        rotation: this.video.poster.rotation,
+        width: this.video.poster.width,
+        height: this.video.poster.height
+      }
+    };
+
+    // Re-fetch the blobs from storage
+    videoReq = this.storage.video.get(video.filepath);
+    videoReq.onerror = onError;
+    videoReq.onsuccess = onVideoSuccess;
+
+    posterReq = this.storage.picture.get(video.poster.filepath);
+    posterReq.onerror = onError;
+    posterReq.onsuccess = onPosterSuccess;
   }
 
   function onError(e) {
-    self.onStopRecordingError(filepath);
+    if (self) {
+      self.onStopRecordingError(video);
+      self = null;
+    }
   }
 
-  function onSuccess() {
+  function onVideoSuccess() {
     debug('got video blob');
-    self.onNewVideo({
-      blob: req.result,
-      filepath: filepath,
-      elapsedTime: elapsedTime
-    });
+    video.blob = videoReq.result;
+    onSuccess();
   }
-};
 
-/**
- * Once we have got the new video blob
- * from storage we assemble the video
- * object and then get video meta data
- * to add to it.
- *
- * If we fail to decode the metadata from
- * the file, we delete it from storage
- * and abort prevent the app from ever
- * knowing a new (potentially corrupted)
- * video file was recorded. If the
- * duration is below the configuration
- * threshold, we will suppress reporting
- * the error, as it is likely due to
- * stopping recording the video very
- * quickly.
- *
- * @param  {Object} video
- * @private
- */
-Camera.prototype.onNewVideo = function(video) {
-  debug('got new video', video);
-  var self = this;
-
-  // Finally extract some metadata before
-  // telling the app the new video is ready.
-  this.getVideoMetaData(video.blob, gotVideoMetaData);
+  function onPosterSuccess() {
+    debug('got poster blob');
+    video.poster.blob = posterReq.result;
+    onSuccess();
+  }
 
-  function gotVideoMetaData(error, data) {
-    if (error) {
-      var tooShort = video.elapsedTime < self.minRecordingTime;
-      self.onStopRecordingError(video.filepath, tooShort);
-      return;
+  function onSuccess() {
+    if (++got == 2) {
```

Comment:
Yeah, my objection here is mostly that it seems fragile that we're counting the number of times get reach a success condition. It seems out-of-scope for this patch to refactor the storage module (although, that would be nice), but @davidflanagan's alternate suggestion seems like it might also work.